cmd/telemetrygen/internal/metrics/metrics.go

```
@@ -0,0 +1,112 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+	"fmt"
+
+	grpcZap "github.com/grpc-ecosystem/go-grpc-middleware/logging/zap"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetricgrpc"
+	"go.opentelemetry.io/otel/exporters/otlp/otlpmetric/otlpmetrichttp"
+	"go.opentelemetry.io/otel/metric/global"
+	sdkmetric "go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/resource"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+)
+
+func Start(cfg *Config) error {
+	logger, err := zap.NewDevelopment()
+	if err != nil {
+		return fmt.Errorf("failed to obtain logger: %w", err)
+	}
+	grpcZap.ReplaceGrpcLoggerV2(logger.WithOptions(
+		zap.AddCallerSkip(3),
+	))
+
+	grpcExpOpt := []otlpmetricgrpc.Option{
+		otlpmetricgrpc.WithEndpoint(cfg.Endpoint),
+		otlpmetricgrpc.WithDialOption(
+			grpc.WithBlock(),
+		),
+	}
+
+	httpExpOpt := []otlpmetrichttp.Option{
+		otlpmetrichttp.WithEndpoint(cfg.Endpoint),
+	}
+
+	if cfg.Insecure {
+		grpcExpOpt = append(grpcExpOpt, otlpmetricgrpc.WithInsecure())
+		httpExpOpt = append(httpExpOpt, otlpmetrichttp.WithInsecure())
+	}
+
+	if len(cfg.Headers) > 0 {
+		grpcExpOpt = append(grpcExpOpt, otlpmetricgrpc.WithHeaders(cfg.Headers))
+		httpExpOpt = append(httpExpOpt, otlpmetrichttp.WithHeaders(cfg.Headers))
+	}
+
+	var exp sdkmetric.Exporter
+	if cfg.UseHTTP {
+		logger.Info("starting HTTP exporter")
+		exp, err = otlpmetrichttp.New(context.Background(), httpExpOpt...)
+	} else {
+		logger.Info("starting gRPC exporter")
+		exp, err = otlpmetricgrpc.New(context.Background(), grpcExpOpt...)
+	}
+
+	if err != nil {
+		logger.Error("failed to obtain OTLP exporter", zap.Error(err))
+		return err
```

Why not just return the wrapped error here?